internal/build/image_and_cache_builder.go

```diff
@@ -0,0 +1,175 @@
+package build
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/distribution/reference"
+	"github.com/windmilleng/tilt/internal/dockerfile"
+	"github.com/windmilleng/tilt/internal/ignore"
+	"github.com/windmilleng/tilt/internal/logger"
+	"github.com/windmilleng/tilt/internal/mode"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/store"
+)
+
+type ImageAndCacheBuilder struct {
+	ib         ImageBuilder
+	cb         CacheBuilder
+	updateMode mode.UpdateMode
+}
+
+func NewImageAndCacheBuilder(ib ImageBuilder, cb CacheBuilder, updateMode mode.UpdateMode) *ImageAndCacheBuilder {
+	return &ImageAndCacheBuilder{
+		ib:         ib,
+		cb:         cb,
+		updateMode: updateMode,
+	}
+}
+
+func (icb *ImageAndCacheBuilder) Build(ctx context.Context, iTarget model.ImageTarget, state store.BuildState, ps *PipelineState, canSkipPush bool) (reference.NamedTagged, error) {
```

Comment:
should canSkipPush be a method of this struct? it seems like a weird thing to be a parameter

Reply:
yeahhh we'd need to mess with the way we k8s env in and how it interacts with Docker Compose -- because we still populate this value even when we're doing Docker Compose stuff instead of worrying about k8s :-/ i'll leave a todo